Samples/Widgets/MultipleWidgets/cpp-win32-packaged/SampleWidgetProviderApp/WidgetProvider.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "WidgetProvider.h"
+#include "WidgetImpl.h"
+
+typedef std::function<std::shared_ptr<WidgetImpl>(winrt::WidgetContext, winrt::hstring)> CreateWidgetImplFn;
+struct WidgetImplCreationInfo
+{
+	CreateWidgetImplFn factoryFn;
+};
+
+template<typename T>
+std::shared_ptr<WidgetImpl> CreateWidgetFn(winrt::WidgetContext widgetContext, winrt::hstring state)
+{
+	auto widgetId = widgetContext.Id();
+	auto newWidget = std::make_shared<T>(widgetId, state);
+
+	// Call UpdateWidget() to send data/template to the newly created widget.
+	winrt::WidgetUpdateRequestOptions updateOptions{ widgetId };
+	updateOptions.Template(newWidget->GetTemplateForWidget());
+	updateOptions.Data(newWidget->GetDataForWidget());
+	// You can store some custom state in the widget service that you will be able to query at any time.
```

Comment:
There's some weird indentation happening in these files. 4 spaces, please, per level.

Reply:
Oh, they're tabs? Please make them spaces.

Reply:
fixed identation everwyhere